clevertap-core/src/main/java/com/clevertap/android/sdk/ManifestInfo.java

```diff
@@ -6,58 +6,89 @@
 import android.os.Bundle;
 import android.text.TextUtils;
 import androidx.annotation.RestrictTo;
-
+import androidx.annotation.VisibleForTesting;
+
+/**
+ * Parser for android manifest and picks up fields from manifest once to be references
+ *
+ * Should be singleton and initialised only once -> need to validate.
+ */
+// todo lp Remove context dependency from here
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 public class ManifestInfo {
```

Comment:
You've changed a lot of code in this class, consider converting it to Kotlin. I think it would benefit the readability.

Reply:
it is introducing change in other files as well, will do in different pr